frontend/src/metabase/query_builder/actions/core/updateQuestion.unit.spec.ts

```diff
@@ -811,6 +814,69 @@ describe("QB Actions > updateQuestion", () => {
     });
   });
 
+  describe("table column sync", () => {
+    it("removes columns from table.columns when a column is removed from a query", async () => {
+      const tableColumns = [
+        createMockTableColumnOrderSetting({
+          name: "ID",
+          fieldRef: ["field", ORDERS.ID, {}],
+          enabled: true,
+        }),
+        createMockTableColumnOrderSetting({
+          name: "TOTAL",
+          fieldRef: ["field", ORDERS.TOTAL, {}],
+          enabled: true,
+        }),
+        createMockTableColumnOrderSetting({
+          name: "QUANTITY",
+          fieldRef: ["field", ORDERS.QUANTITY, {}],
```

Comment:
I'm not sure we can have field references looking like that

Usually, they either have at least one option:

```js
[ "field", ORDERS.CREATED_AT, { "temporal-unit": "month" } ]

[ "field", "TOTAL", { "base-type": "type/Float" } ]
```

or no options at all:

```js
[ "field", ORDERS.TOTAL, null ]
```

Could we please replace `{}` with `nulls` here so it's a bit closer to reality?

Reply:
Yeah, this felt a bit weird. I think I did this for the type check, but I'll take another look and see if I can sort it out 👍